strapi/strapi
# Solve foreign key error on push transfers

### What does it do?

It solves the foreign key error on push transfers by fixing the way we are getting the component UIDs when mapping the old IDs and the new IDs.

### Why is it needed?

To avoid FK Constraint errors on transfers. 

### How to test it?

Create a content type with a component and a repeatable component, both containing relation to another content type inside. Fill them with data and execute an export/import and/or a transfer. It should work without displaying any warning about a foreign key constraint. 

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/18350
